test/src/example.cpp

```diff
@@ -103,6 +112,16 @@ void Example::emit_custom_signal(const String &name, int value) {
 	emit_signal("custom_signal", name, value);
 }
 
+Array Example::test_array() const {
+	Array arr;
+
+	arr.resize(2);
+	arr[0] = Variant(1);
```

Comment:
@vnen so this test case currently crashes but seemingly not in relation to the array but the way the variants are accessed. 

Any chance you could try out my PR and see the crash for yourself? I haven't got a clue what is going on here.